pkgs/tools/package-management/apkg/default.nix

```
@@ -4,47 +4,38 @@
 
 python3Packages.buildPythonApplication rec {
   pname = "apkg";
-  version = "0.4.0";
+  version = "0.4.1";
+  format = "pyproject";
 
   src = fetchFromGitLab {
     domain = "gitlab.nic.cz";
     owner = "packaging";
     repo = pname;
     rev = "v${version}";
-    sha256 = "duZz2Kwjgek5pMJTDH8gMZAZ13uFwaIYT5E1brW7I7U=";
+    sha256 = "x7UYkqkF1XJ3OMfQpIQ4+27KI0dLvL42Wms5xQTY/H4=";
   };
 
-  # TODO: solve this properly.  Detection won't work anymore.
-  postPatch = ''
-    patch <<-EOF
-      --- a/setup.py
-      +++ b/setup.py
-      @@ -25,1 +25,1 @@
-      -    version=version,
-      +    version='${version}',
-    EOF
-  '';
-
   propagatedBuildInputs = with python3Packages; [
     # copy&pasted requirements.txt (almost exactly)
     beautifulsoup4   # upstream version detection
-    blessings        # terminal colors
+    blessed          # terminal colors
     build            # apkg distribution
-    cached-property  # @cached_property for python <= 3.7
+    cached-property  # for python <= 3.7; but pip complains even with 3.8
     click            # nice CLI framework
     distro           # current distro detection
     jinja2           # templating
     packaging        # version parsing
     requests         # HTTP for humans™
-    setuptools       # required by minver
     toml             # config files
+
+    hatchling
   ];
```

`hatchling` should be in `nativeBuildInputs` since it's only required at build time
```suggestion
  ];
  
  nativeBuildInputs = with python3Packages; [ hatchling ];
```

---

Right, amended.  I don't expect cross-compilation but it will be better.